action_server_bt/include/action_server_bt/action_server_bt.hpp

```diff
@@ -36,14 +36,18 @@ class ActionServerBT
   using ActionTree = action_server_bt_msgs::action::ActionTree;
   using GoalHandleActionTree = rclcpp_action::ServerGoalHandle<ActionTree>;
 
+  typedef std::function<void(BT::Tree&)> OnTreeCreatedCallback;
+  typedef std::function<void(BT::NodeStatus&)> OnTreeExecutionCompletedCallback;
+
   /**
    * @brief Constructor for ActionServerBT.
    * @details This initializes a ParameterListener to read configurable options from the user and
    * starts an Action Server that takes requests to execute BehaviorTrees.
    *
    * @param options rclcpp::NodeOptions to pass to node_ when initializing it.
    */
-  explicit ActionServerBT(const rclcpp::NodeOptions& options);
+  explicit ActionServerBT(const rclcpp::NodeOptions& options, OnTreeCreatedCallback tree_created,
+                          OnTreeExecutionCompletedCallback execution_complete);
```

Comment:
```suggestion
  explicit ActionServerBT(const rclcpp::NodeOptions& options, OnTreeCreatedCallback tree_created,
                          OnTreeExecutionCompletedCallback execution_complete);
                          
   explicit ActionServerBT(const rclcpp::NodeOptions& options):
     ActionServerBT(options, OnTreeCreatedCallback{}, OnTreeExecutionCompletedCallback{}) 
   {}                               
```

Reply:
I believe that a good number of users will not need to specify these callbacks. Can we provide a specific constructor that is less verbose?

Reply:
All the options are available now :smiley:

Reply:
Okay I didn't like all the options after I added the `on_loop` option so I collected them into a struct with default initializes. Let me know what you think about this approach.